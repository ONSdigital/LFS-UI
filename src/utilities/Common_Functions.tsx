function weeks() {
    let i;
    let weekList = []
    for (i=1; i<=52; i++) {
        weekList.push({"label":"Week "+String(i), "value":String(i)})
    }
    return weekList
}

function months() {
    const monthNames = ["January", "February", "March", "April", "May", "June", "July", 
                        "August", "September", "October", "November", "December"];
    let i;
    let monthlist = [];
    var date = new Date();
    const month = date.getMonth();
    for (i=0; i<12; i++){
        if (month-i>=0)
            monthlist.push({"label":monthNames[month-i], "value":monthNames[month-i]})
        else{
            monthlist.push({"label":monthNames[month+12-i], "value":monthNames[month+12-i]})

        }
    }
    return monthlist
}

const quarters = [{"label":"Q1", "value":"Q2"}, 
                  {"label":"Q2", "value":"Q2"}, 
                  {"label":"Q3", "value":"Q3"}, 
                  {"label":"Q4", "value":"Q4"}]

function years() {
    let i;
    let yearlist = [];
    var date = new Date(); 
    const year = date.getFullYear();
    for (i=0; i<10; i++){
        yearlist.push({"label":String(year-i), "value":String(year-i)})
    }
    return yearlist
}

function getMonth (Batch_ID: String) {
    var Month = Batch_ID.substring(0,3)
    switch (Month){
      case 'Jan': Month = 'January'; break;
      case 'Feb': Month = 'February'; break;
      case 'Mar': Month = 'March'; break;
      case 'Apr': Month = 'April'; break;
      case 'May': break;
      case 'Jun': Month = 'June'; break;
      case 'Jul': Month = 'July'; break;
      case 'Aug': Month = 'August'; break;
      case 'Sep': Month = 'September'; break;
      case 'Oct': Month = 'October'; break;
      case 'Nov': Month = 'November'; break;
      case 'Dec': Month = 'December'; break;
    }
    return Month
  }

  function getYear (Batch_ID: String) {
    return Batch_ID.substring(3,7)
  }

const batches = [{"label":"Monthly", "value":"monthly"}, 
        {"label":"Quarterly", "value":"quarterly"}, 
        {"label":"Yearly", "value":"yearly"}]

const qList = ["January", "February", "April", "May", "July", "August", "October", "November"]

export{weeks, months, quarters, years, batches, getMonth, getYear, qList}