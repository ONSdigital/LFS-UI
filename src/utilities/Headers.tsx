const BATCH_HEADERS =
    [{
        label: "Source",
        column_name: "type",
        filter: false,
        order: false
    }, {
        label: "Period",
        column_name: "month",
        filter: false,
        order: false
    }, {
        label: "Status",
        column_name: "status",
        filter: false,
        order: false
    }, {
        label: "",
        column_name: "button",
        filter: false,
        order: false
    }, {
        label: "",
        column_name: "button",
        filter: false,
        order: false
    }];

const REFERENCE_FILE_HEADERS =
    [{
        label: "File Name",
        column_name: "File Name",
        filter: false,
        order: false
    }, {
        label: "Import Date",
        column_name: "Import Date",
        filter: false,
        order: true
    }, {
        label: "Status",
        column_name: "status",
        filter: false,
        order: false
    }];

const REFERENCE_FILE_IMPORT_HEADERS =
    [{
        label: "File Name",
        column_name: "File Name",
        filter: false,
        order: false
    }, {
        label: "Import Date",
        column_name: "Import Date",
        filter: false,
        order: true
    }, {
        label: "Status",
        column_name: "status",
        filter: false,
        order: true
    },{
        label: "",
        column_name: "",
        filter: false,
        order: false
    }];

const UPLOAD_HEADERS =
    [{
        label: "File Name ",
        column_name: "fileName",
        filter: false,
        order: false
    }, {
        label: "Step",
        column_name: "step",
        filter: false,
        order: false
    }, {
        label: "Status",
        column_name: "status",
        filter: false,
        order: false
    }];

const DASHBOARD_HEADERS =
    [{
        label: "BatchID",
        column_name: "id",
        filter: false,
        order: true
    }, {
        label: "Batch Type",
        column_name: "type",
        filter: false,
        order: true
    }, {
        label: "Period",
        column_name: "period",
        filter: false,
        order: true
    }, {
        label: "Year",
        column_name: "year",
        filter: false,
        order: true
    }, {
        label: "Status",
        column_name: "status",
        filter: false,
        order: true
    }
    ];

const BATCH_PROGRESS_TABLE =
    [{
        label: "Step",
        column_name: "Step",
        filter: false,
        order: true
    }, {
        label: "Status",
        column_name: "status",
        filter: false,
        order: false
    }
    ];

const SURVEY_UPLOAD_HISTORY =
    [{
        label: "Import Date",
        column_name: "Date",
        filter: false,
        order: true
    }, {
        label: "Status",
        column_name: "Status",
        filter: false,
        order: true
    }
    ];

const VARIABLE_DEFINITION_HEADERS =
    [{
        label: "Variable",
        column_name: "variable",
        filter: false,
        order: false
    }, {
        label: "Description",
        column_name: "description",
        filter: false,
        order: false
    }, {
        label: "Valid From",
        column_name: "validFrom",
        filter: false,
        order: false
    }, {
        label: "Editable",
        column_name: "editable",
        filter: false,
        order: false
    }, {
        label: "Imputation",
        column_name: "imputation",
        filter: false,
        order: false
    }, {
        label: "DV",
        column_name: "dv",
        filter: false,
        order: false
    }, {
        label: "Type",
        column_name: "type",
        filter: false,
        order: false
    }, {
        label: "Length",
        column_name: "length",
        filter: false,
        order: false
    }
    ];

const VALUE_LABELS_HEADERS =
    [{
        label: "Variable Name",
        column_name: "Variable Name",
        filter: false,
        order: false
    },{
        label: "Label Name",
        column_name: "Label Name",
        filter: false,
        order: false
    }, {
        label: "Value",
        column_name: "Value",
        filter: false,
        order: false
    }, {
        label: "Description",
        column_name: "Value",
        filter: false,
        order: false
    }, {
        label: "Last Updated",
        column_name: "last Updated",
        filter: false,
        order: false
    }
    ];

function userHeaders() {
    return (
        [{
            label: "User_ID",
            column_name: "user_id",
            filter: false,
            order: true,
            create: false,
        }, {
            label: "Username",
            column_name: "username",
            filter: false,
            order: true,
            create: true
        }, {
            label: "Password",
            column_name: "password",
            filter: false,
            order: false,
            create: true
        }, {
            label: "Role",
            column_name: "role",
            filter: false,
            order: true,
            create: true
        }]
    )
}

function roleHeaders() {
    return (
        [{
            label: "Role Name",
            column_name: "Name",
            filter: false,
            order: true
        }, {
            label: "Page Access",
            column_name: "Pages",
            filter: false,
            order: false
        }]
    )
}

export {BATCH_HEADERS, SURVEY_UPLOAD_HISTORY, UPLOAD_HEADERS, userHeaders, roleHeaders,
    DASHBOARD_HEADERS, VARIABLE_DEFINITION_HEADERS, REFERENCE_FILE_IMPORT_HEADERS, VALUE_LABELS_HEADERS, REFERENCE_FILE_HEADERS, BATCH_PROGRESS_TABLE}