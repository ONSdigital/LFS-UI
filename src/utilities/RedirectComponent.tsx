import {useHistory} from "react-router-dom";
import {useEffect} from "react";

interface Props {
    url: string
}

export function RedirectComponent(props : Props) {
    let history = useHistory();

    useEffect(() => {
        history.push(props.url);
    });
}