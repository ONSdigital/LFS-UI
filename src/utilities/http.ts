function postSurveyFile(lfsfile: any, fileName: string, fileType: string, survey: string, period: string, year: string): Promise<any> {

    let formData = new FormData();
    formData.append("lfsFile", lfsfile[0]);
    formData.append("fileName", fileName);
    formData.append("fileSource", survey);

    return new Promise((resolve: any, reject: any) => {
        fetch("/imports/" + fileType + "/" + survey + "/" +  period + "/" +  year, {
            "method": "POST",
            "body": formData,
        })
            .then(response => {
                console.log(response);
                resolve(response.json());
            })
            .catch(err => {
                console.log(err);
                reject(err)
            });
    })
}

function postImportFile(importFile: any, link :string, fileName: string): Promise<any> {
    console.log("/imports/" + fileName + "-----------------------");
    let formData = new FormData();
    formData.append("lfsFile", importFile[0]);
    formData.append("fileName", fileName);

    return new Promise((resolve: any, reject: any) => {
        fetch("/imports/" + link, {
            "method": "POST",
            "body": formData,
        })
            .then(response => {
                console.log(response);
                resolve(response.json());
            })
            .catch(err => {
                console.log(err);
                reject(err)
            });
    })
}

function createNewBatch(batchType: string, year: string, period: string, description: string): Promise<any> {

    let formData = new FormData();
    formData.append("description", description);

    let url = "/batches/" + batchType + "/" + year + (batchType !== 'annually' ? "/" + period : "");

    return new Promise((resolve: any, reject: any) => {
        fetch(url, {
            "method": "POST",
            "body": formData,
        })
            .then(response => {
                console.log("Response");
                console.log(response);
                resolve(response.json());
            })
            .catch(err => {
                console.log("Error");
                console.log(err);
                reject(err)
            });
    })
}

function getBatchData(batchType: string, year: string, period: string): Promise<any> {
    let url = "/batches/display/" + batchType + "/" + year + (batchType !== 'annually' ? "/" + period : "");

    return new Promise((resolve: any, reject: any) => {
        fetch(url, {
            "method": "GET"
        })
            .then(response => {
                console.log("Response");
                console.log(response);
                resolve(response.json());
            })
            .catch(err => {
                console.log("Error");
                console.log(err);
                reject(err)
            });
    })
}

function getAllBatches(): Promise<any> {
    return new Promise((resolve: any, reject: any) => {
        fetch("/dashboard", {
            "method": "GET"
        })
            .then(response => {
                console.log("Response");
                console.log(response);
                resolve(response.json());
            })
            .catch(err => {
                console.log("Error");
                console.log(err);
                reject(err)
            });
    })
}

export function getAllVarDefs(): Promise<any> {
    return new Promise((resolve: any, reject: any) => {
        fetch("/variable/definitions", {
            "method": "GET"
        })
            .then(response => {
                console.log("Response");
                console.log(response);
                resolve(response.json());
            })
            .catch(err => {
                console.log("Error");
                console.log(err);
                reject(err)
            });
    })
}

function getSurveyAudit(survey: string, year: string, period: string): Promise<any> {
    let url = "/audits/" + (survey === 'gb' ? 'week' : 'month') + "/" + year + "/" + period;

    return new Promise((resolve: any, reject: any) => {
        fetch(url, {
            "method": "GET"
        })
            .then(response => {
                console.log("Response");
                console.log(response);
                resolve(response.json());
            })
            .catch(err => {
                console.log("Error");
                console.log(err);
                reject(err)
            });
    })
}

export {postSurveyFile, postImportFile, createNewBatch, getAllBatches, getBatchData, getSurveyAudit}
