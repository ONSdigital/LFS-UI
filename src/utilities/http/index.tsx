export * from "./survey";
export * from "./http";
export * from './batch';