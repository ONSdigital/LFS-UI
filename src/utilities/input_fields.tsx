function weeks() {
    let i;
    let weekList = []
    for (i=1; i<=52; i++) {
        weekList.push({"label":"Week "+String(i), "value":String(i)})
    }
    return weekList
}

function months() {
    const monthNames = ["January", "February", "March", "April", "May", "June", "July", 
                        "August", "September", "October", "November", "December"];
    let i;
    let monthlist = [];
    var date = new Date();
    const month = date.getMonth();
    for (i=0; i<12; i++){
        if (month-i>=0)
            monthlist.push({"label":monthNames[month-i], "value":monthNames[month-i]})
        else{
            monthlist.push({"label":monthNames[month+12-i], "value":monthNames[month+12-i]})

        }
    }
    return monthlist
}

const quarters = [{"label":"Q1", "value":"Q2"}, 
                  {"label":"Q2", "value":"Q2"}, 
                  {"label":"Q3", "value":"Q3"}, 
                  {"label":"Q4", "value":"Q4"}]

function years() {
    let i;
    let yearlist = [];
    var date = new Date(); 
    const year = date.getFullYear();
    for (i=0; i<10; i++){
        yearlist.push({"label":String(year-i), "value":String(year-i)})
    }
    return yearlist
}

const runs = [{"label":"Monthly", "value":"monthly"}, 
        {"label":"Quarterly", "value":"quarterly"}, 
        {"label":"Yearly", "value":"yearly"}]

export{weeks, months, quarters, years, runs}