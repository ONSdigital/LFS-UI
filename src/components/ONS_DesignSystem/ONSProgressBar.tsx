import React from 'react';
import {getStatusStyle} from "../../utilities/Common_Functions";

interface Props {
    statusCode: number
    percentage: number
}

interface FillerProps {
    percentage: number
    statusCode: number
}

export const ONSProgressBar = (props: Props) => (
    <p className="field">
        <p className="field">
            <div className={'progress-bar'}>
                <Filler percentage={props.percentage} statusCode={props.statusCode}/>
            </div>
        </p>
    </p>
);

const Filler = (props: FillerProps) => {
    return <div className="progress-bar-filler" style={{width: `${props.percentage}%`, background: `${getStatusStyle(props.statusCode).hexCode}`}}/>
};