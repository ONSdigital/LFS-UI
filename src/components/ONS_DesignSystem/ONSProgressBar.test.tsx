import React from "react";
import Enzyme, {mount, shallow} from "enzyme";
import {ONSProgressBar} from "./ONSProgressBar";
import Adapter from "enzyme-adapter-react-16";
import {getUploadStatusStyle} from "../../utilities/Common_Functions";

describe("ONS Progress Bar Test", () => {
    Enzyme.configure({ adapter: new Adapter() });

    const Props = {
        statusCode: 1,
        percentage: 69
    }

    function wrapper(render: any, props: any) {
        return render(
            <ONSProgressBar
                statusCode={props.statusCode}
                percentage={props.percentage}
            />
            
        )
    }

    it("matches Snapshot", () => {
        expect(wrapper(shallow, Props)).toMatchSnapshot()
    });

    it("should render correctly", () => expect(wrapper(shallow, Props).exists()).toEqual(true));

    it("shows the correct status", () => {
        let wrapperStyle = wrapper(mount, Props).find("div.progress-bar-filler").prop('style')
        expect(wrapperStyle).toHaveProperty('background', getUploadStatusStyle(Props.statusCode).hexCode);
    })

    it('shows the correct percentage on the bar', () => {
        let wrapperStyle = wrapper(mount, Props).find("div.progress-bar-filler").prop('style')
        expect(wrapperStyle).toHaveProperty('width', Props.percentage + "%");
    })

})