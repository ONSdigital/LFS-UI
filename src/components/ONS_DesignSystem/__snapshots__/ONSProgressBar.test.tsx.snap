// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ONS Progress Bar Test matches Snapshot 1`] = `ShallowWrapper {}`;
